src/common/meta/src/key.rs

```diff
@@ -859,12 +880,19 @@ impl TableMetadataManager {
         table_id: TableId,
         table_name: &TableName,
         table_route_value: &TableRouteValue,
+        region_wal_options: &HashMap<RegionNumber, String>,
     ) -> Result<()> {
         let keys = self.table_metadata_keys(table_id, table_name, table_route_value)?;
```

Comment:
How about making `self.table_metadata_keys` accept the `region_wal_options` argument, and return all keys that need to be deleted?